src/main/java/io/druid/data/input/impl/SpatialDimensionSchema.java

```diff
@@ -18,12 +18,14 @@
 package io.druid.data.input.impl;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.List;
 
 /**
  */
+@Deprecated
 public class SpatialDimensionSchema
 {
   private final String dimName;
```

Comment:
@himanshug 

Ultimately, I think this should be the way users express spatial dimensions in the spec:
https://github.com/druid-io/druid/issues/2622

However, it will take a while for selectable indexes to be supported, so I didn't want to potentially confuse users by going that route immediately.

Going forward, with this patch as-is, the user can continue to use the old convention of defining spatials in a separate section, or they could use the "NewSpatialDimensionSchema" convention in the main "dimensions" list. 

When configurable indexes are supported, then I plan on introducing the changes in issue #2622 above. At that point, or at the soonest release point where we can make backwards incompatible changes, we can drop the old spatial dimension schema, and either drop the "NewSpatialDimensionSchema" or deprecate it.